lib/dio/forwarders/attribute_forwarder.rb

```diff
@@ -0,0 +1,71 @@
+require 'delegate'
+
+module Dio
+  module Forwarders
+    # A more controlled forwarder that targets only `attr_` type methods like
+    # `attr_reader` and `attr_accessor`. These attributes are found by diffing
+    # instance variables and method names to find generated accessor methods.
+    #
+    # It should be noted that this could be faster if there was an assumption
+    # of purity at the time of the match. I may make another variant for that.
+    #
+    # @author [baweaver]
+    # @since 0.0.3
+    #
+    class AttributeForwarder < BaseForwarder
+      def initialize(base_object)
+        ivars = Set.new base_object
+          .instance_variables
+          .map { _1.to_s.delete('@').to_sym }
+
+        all_methods = Set.new base_object.methods
+
+        @attributes = ivars.intersection(all_methods)
+
+        super
+      end
+
+      # Deconstructs from a list of attribute names, wrapping each value
+      # in a new dive in case the pattern match goes further than one level.
+      #
+      # @return [Array]
+      def deconstruct
+        @attributes.map { NEW_DIVE[@base_object.send(_1)] }
```

Comment:
Curious if #public_send suffices for these? Assuming no, or you'd have used it. :)